.github/workflows/ci.yml

```diff
@@ -154,34 +130,33 @@ jobs:
           args: --all -- --check
 
   wasm-tests:
-    name: WASM tests
-    runs-on: ${{matrix.os}}-latest
+    name: ${{matrix.name}}
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        toolchain: [stable]
-        os: [ubuntu]
+        include:
+          - name: Build WASM compatible binary
+            command: wasm-pack build --features wasm
+
+          - name: Test wasm-pack with default profile (blake3)
+            command: wasm-pack test --node --features wasm
+
+          - name: Test wasm-pack with SHA3_256 and no default features
+            command: wasm-pack test --node --no-default-features --features wasm,sha3_256
 
     steps:
       - uses: actions/checkout@main
 
       - name: Install rust
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: ${{matrix.toolchain}}
+          toolchain: stable
           override: true
 
       - name: Install wasm-pack
         run: curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
 
-      - name: Build WASM compatible binary
-        working-directory: ./akd_client
-        run: wasm-pack build --features wasm
-
-      - name: Test default profile (blake3)
-        working-directory: ./akd_client
-        run: wasm-pack test --node --features wasm
-
-      - name: Test SHA3_256 and no default features
+      - name: Run test
         working-directory: ./akd_client
-        run: wasm-pack test --node --no-default-features --features wasm,sha3_256
\ No newline at end of file
+        run: ${{matrix.command}}
```

Comment:
nit: could this be something like 

```
wasm-pack ${{matrix.command}} ${{matrix.args}}
```

then the strategy would be a little more standardized with the previous block?

Reply:
whoops, merged without thinking. Maybe i'll clean this up later, sorry!